src/AWS.Deploy.CLI/Program.cs

```diff
@@ -2,245 +2,31 @@
-                        ProjectDefinition = projectDefinition,
-                        SystemCapabilities = systemCapabilities
-                    };
-
-                    var awsResourceQueryer = new AWSResourceQueryer(awsClientFactory);
-                    var zipFileManager = new ZipFileManager(commandLineWrapper);
-                    var consoleUtilities = new ConsoleUtilities(toolInteractiveService);
-
-                    var deploy = new DeployCommand(
-                        toolInteractiveService,
-                        orchestratorInteractiveService,
-                        new CdkProjectHandler(orchestratorInteractiveService, commandLineWrapper),
-                        cdkManager,
-                        new DeploymentBundleHandler(commandLineWrapper, awsResourceQueryer, orchestratorInteractiveService, directoryManager, zipFileManager),
-                        new DockerEngine.DockerEngine(projectDefinition),
-                        awsResourceQueryer,
-                        new TemplateMetadataReader(awsClientFactory),
-                        new DeployedApplicationQueryer(awsResourceQueryer),
-                        new TypeHintCommandFactory(toolInteractiveService, awsResourceQueryer, consoleUtilities),
-                        new CloudApplicationNameGenerator(fileManager, directoryManager),
-                        consoleUtilities,
-                        session);
-
-                    await deploy.ExecuteAsync(stackName, saveCdkProject);
-
-                    return CommandReturnCodes.SUCCESS;
-                }
-                catch (Exception e) when (e.IsAWSDeploymentExpectedException())
-                {
-                    // helpful error message should have already been presented to the user,
-                    // bail out with an non-zero return code.
-                    return CommandReturnCodes.USER_ERROR;
-                }
-                catch (Exception e)
-                {
-                    // This is a bug
-                    toolInteractiveService.WriteErrorLine(
-                        "Unhandled exception.  This is a bug.  Please copy the stack trace below and file a bug at https://github.com/aws/aws-dotnet-deploy. " +
-                        e.PrettyPrint());
-
-                    return CommandReturnCodes.UNHANDLED_EXCEPTION;
-                }
-            });
-            rootCommand.Add(deployCommand);
-
-            var listCommand = new Command("list-deployments", "List existing deployments.")
-            {
-                _optionProfile,
-                _optionRegion,
-                _optionProjectPath,
-                _optionDiagnosticLogging
-            };
-            listCommand.Handler = CommandHandler.Create<string, string, string, bool>(async (profile, region, projectPath, diagnostics) =>
+            // calls the Run method in App, which is replacing Main
+            var app = serviceProvider.GetService<App>();
+            if (app == null)
```

Comment:
nit:  Rather than using a run time exception to verify bindings use a unit test:

```
[Fact]
public void ServiceBindingsCanBuildTheApp()
{
    new ServiceCollection()
        .AddCustomServices()
        .BuildServiceProvider()
        .GetService<App>()
        .ShouldNotBeNull();
}
```